### PR TITLE
fix(fs): guard mount-recovery lick and give cone an actionable prompt

### PR DESCRIPTION
**Pull request by trieloff**
## Summary

Fixes three issues with mount-point recovery after a page reload:

1. **Documents when recovery is required.** Chrome's File System Access API only retains readwrite permission inside a tab's active session, so a soft navigation usually keeps `queryPermission` at `granted` while a full browser restart drops handles back to `prompt`. The new `fs/mount-recovery.ts` header explains this, and the comment in `main.ts` points readers there so the behaviour is no longer surprising.

2. **Only fires the `session-reload` lick when recovery is actually needed.** The recovery logic now lives in a pure helper `recoverMounts(entries, fs)` that returns a `needsRecovery` bucket. `main.ts` early-returns when the bucket is empty, and `routeLickToScoop` defensively drops any `mount-recovery` event with an empty mounts list so the cone is never pestered with an empty notification.

3. **Gives the cone an actionable prompt instead of a raw JSON dump.** `formatMountRecoveryPrompt` produces a natural-language message that tells the cone to inform the user, lists the affected paths (with the original directory name when available), and offers the exact `mount <path>` / `mount unmount <path>` commands needed to re-authorize each one.

## Tests

New `packages/webapp/tests/fs/mount-recovery.test.ts` (12 tests) covers:

- `granted` / `prompt` / `denied` permission states
- handles missing `queryPermission` entirely (legacy records)
- handles whose `queryPermission` throws
- `fs.mount()` itself failing after permission check
- mixed restored + needs-recovery runs, empty input
- prompt formatter: null-on-empty contract, pluralisation, omission of dirName when unknown, inclusion of `mount` and `mount unmount` commands

## Verification

```
npx prettier --write <changed-files>    # clean
npm run typecheck                        # passes
npm run test                             # 2341 passing (pre-existing 4 unrelated failures unchanged)
npm run build                            # passes
```

## Non-changes

- No changes to the persistence layer in `mount-table-store.ts` — the recovery fix is purely on the consumption side.
- No user-facing docs updated; the new module header and inline comment in `main.ts` are where anyone debugging reload behaviour will look first.

### CODE REVIEW

File: packages/webapp/src/fs/mount-recovery.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+ * a `null` result as "do not emit a lick".
+ */
+export function formatMountRecoveryPrompt(mounts: MountRecoveryEntry[]): string | null {
+  if (!Array.isArray(mounts) || mounts.length === 0) return null;
+
+  const listLines = mounts.map(({ path, dirName }) => {
+    const origin = dirName ? ` (previously mounted from \`${dirName}\`)` : '';
+    return `- \`${path}\`${origin}`;
+  });
+  const mountCmds = mounts.map(({ path }) => `    mount ${path}`);
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/fs/mount-recovery.ts]: Fixed in 31f22bf by routing mount commands through a new `shellQuote` helper (POSIX single-quote form, `'\''` escape for embedded quotes). Added tests for paths with spaces, single quotes, and the empty string. The suggested line now renders as `mount '/mnt/My Project'`.

File: packages/webapp/src/fs/mount-recovery.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/fs/mount-recovery.ts:172]: Fixed in 31f22bf — mount commands are now shell-quoted via `shellQuote` (single-quote form). Same helper will be reused if we ever emit per-path `unmount` commands, but for now the prompt points the cone at the manual `mount unmount <path>` form with an explicit note to quote the path the same way.

File: packages/webapp/src/fs/mount-recovery.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/src/fs/mount-recovery.ts]: Fixed in 31f22bf. Went with a stronger approach than the suggested \\`` escape (which isn't actually valid inside Markdown inline code per CommonMark §6.1): the new `mdInlineCode` helper picks a delimiter run that's longer than any backtick sequence in the value, collapses embedded CR/LF to a single space, and pads leading/trailing backticks with a space. So `/mnt/weird\`path` renders as `\`\`/mnt/weird\`path\`\`` and renderers parse it cleanly.

File: packages/webapp/tests/fs/mount-recovery.test.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  it('pluralizes when multiple mounts need recovery', () => {
+    const prompt = formatMountRecoveryPrompt([
+      { path: '/workspace/a', dirName: 'a' },
+      { path: '/workspace/b', dirName: 'b' },
+    ]);
+    expect(prompt).not.toBeNull();
+    expect(prompt).toContain('2 mount points');
+    expect(prompt).toContain('mount /workspace/a');
+    expect(prompt).toContain('mount /workspace/b');
+  });
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/webapp/tests/fs/mount-recovery.test.ts:181]: Fixed in 31f22bf. Added a dedicated `shell-quotes mount paths containing spaces so they parse as one argv token` test for `/mnt/My Project`, plus a companion test for embedded single quotes (`/mnt/It's Work`). Also added standalone unit tests for `shellQuote` covering plain / space / single-quote / quote-only / empty inputs, and for `mdInlineCode` covering the delimiter-growth and leading/trailing-backtick padding rules.
